io/src/ply_io.cpp

```diff
@@ -267,7 +267,8 @@ namespace pcl
   template<typename Scalar> void
   PLYReader::vertexScalarPropertyCallback (Scalar value)
   {
-    if (!std::isfinite (value))
+    //MSVC is missing bool std::isfinite(IntegralType arg); variant, so we need to cast ourself to double
+    if (!std::isfinite (static_cast<double>(value)))
```

Comment:
What about if we add a local helper function, something like

```cpp
template<typename T> inline void
unsetDenseFlagIfNotFinite(T value, PCLPointCloud2& cloud)
{
  if (!std::isfinite)
    cloud.is_dense = false;
}
```

with a specialization for integral types that is a no-op?

Reply:
10 lines of code vs a single cast, but okay, I used template specialization for it ;).

Reply:
Sorry for the trouble :)